src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts

```diff
@@ -0,0 +1,136 @@
+
+
+@Component({
+  selector: 'app-create-collection',
+  templateUrl: './create-collection.component.html',
+  styleUrls: ['./create-collection.component.css']
+})
+export class CreateCollectionComponent implements OnInit {
+
+  isCollectionCreated: boolean;
+
+  /**
+	 * This variable hepls to show and hide page loader.
+   * It is kept true by default as at first when we comes
+   * to a page the loader should be displayed before showing
+   * any data
+	 */
+  showLoader = true;
+  /**
+  * To show toaster(error, success etc) after any API calls
+  */
+  private toasterService: ToasterService;
+  /**
+   * To call resource service which helps to use language constant
+   */
+  public userService: UserService;
+  /**
+    * To call resource service which helps to use language constant
+    */
+  public resourceService: ResourceService;
+  /**
+   * To make inbox API calls
+   */
+  public userForm: FormGroup;
+
+  private editorService: EditorService;
+  public state: string;
+  public type: string;
+  public framework: string;
+  public showEditor: boolean;
+  public contentId: any;
+  public requestBody: any;
+  public userProfile: any;
+
+
+  public collection = new collectionDataInterface('', '');
+  constructor(public modalService: SuiModalService,
+    resourceService: ResourceService,
+    toasterService: ToasterService,
+    editorService: EditorService,
+    private router: Router,
+    userService: UserService,
+    public formBuilder: FormBuilder
+  ) {
+    this.resourceService = resourceService;
+    this.toasterService = toasterService;
+    this.editorService = editorService;
+    this.userService = userService;
+
+    this.userForm = this.formBuilder.group({
+      name: '',
+      description: ''
+    });
+  }
+
+  ngOnInit() {
+    this.userService.userData$.subscribe(
+      (user: IUserData) => {
+        if (user && !user.err) {
+          this.userProfile = user.userProfile;
+          console.log(' user profile s', this.userProfile);
+        }
+      });
+    this.showLoader = false;
+  }
+
+
+
+  saveMetaData(userFormData) {
+    this.showLoader = true;
+    this.requestBody = userFormData;
+    this.requestBody = {
+      name: this.requestBody.name ? this.requestBody.name : 'Untitled Collection',
+      description: this.requestBody.description,
+      creator: this.userProfile.firstName + ' ' + this.userProfile.lastName,
+      createdBy: this.userProfile.id,
+      organisationIds: this.userProfile.organisationIds,
+      createdFor: this.userProfile.organisationIds,
+      userRoles: this.userProfile.userRoles,
+      mimeType: 'application/vnd.ekstep.content-collection',
+      contentType: 'Collection'
+    };
+
+    const requestData = {
+      content: this.requestBody
+    };
+    this.createCollection(requestData);
+  }
+
+
+  createCollection(requestData) {
+    this.editorService.create(requestData).subscribe(res => {
+      this.isCollectionCreated = true;
```

Comment:
why we need this.isCollectionCreated flag. One success we are redirecting to a different page. If failed toaster message is need.

Reply:
For showing the loader inside the modal, but now i am hiding the modal onclick of the button